### PR TITLE
Improved storybook performance & migrate to forwardRefSimple

### PR DESCRIPTION
**Pull request by anuraghazra**

### CODE REVIEW

File: src/button/IconButton.tsx [hunk 1]
```diff
@@ -18,10 +18,11 @@ export type IconButtonProps = ButtonProps & {
   variant?: "primary" | "secondary" | "outline" | "ghost" | AnyString;
 };
 
-function IconButtonComponent(
-  props: PropsWithAs<IconButtonProps, "button">,
-  ref: React.Ref<HTMLButtonElement>,
-) {
+const IconButton = forwardRefWithAsSimple<
```
Comment (navin-moorthy) [inline src/button/IconButton.tsx:21]: Needed to add `export`

Comment (navin-moorthy) [reply src/button/IconButton.tsx:21]: I added them and will push it.

File: src/utils/types.tsx [hunk 2]
```diff
@@ -14,6 +14,12 @@ export type ComponentWithAs<Props, DefaultType extends As> = {
   (props: PropsWithAs<Props, DefaultType>): JSX.Element;
 };
 
+/**
+ * @template Props Component Props
+ * @template RefProp HTML intrinsic type, eg: HTMLDivElement
+ * @template DefaultType string, eg: "div"
+ * @param component
+ */
```
Comment (navin-moorthy) [inline src/utils/types.tsx:22]: I saw them in the twitter !!
Using `@template`

Comment (anuraghazra) [reply src/utils/types.tsx:22]: Yeah.. its actually officially documented but never noticed..
https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html

**Review by navin-moorthy — APPROVED** [review]
LGTM 🎉

**Comment by navin-moorthy** [comment]
Just remove the old forwardRef & rename `simple` so we use the previous name.

**Pull request merged**